ComponentKit/Core/CKComponentController.mm

```diff
@@ -191,15 +193,31 @@ - (void)_relinquishView
 - (void)_cleanupAppliedAnimations
 {
   for (const auto &appliedAnimation : _appliedAnimationsOnInitialMount) {
-    appliedAnimation.animation.cleanup(appliedAnimation.context);
+    appliedAnimation.second.animation.cleanup(appliedAnimation.second.context);
   }
   _appliedAnimationsOnInitialMount.clear();
   for (const auto &appliedAnimation : _appliedAnimations) {
-    appliedAnimation.animation.cleanup(appliedAnimation.context);
+    appliedAnimation.second.animation.cleanup(appliedAnimation.second.context);
   }
   _appliedAnimations.clear();
 }
 
+- (void)_applyAnimation:(const CKPendingComponentAnimation &)pendingAnimation toMap:(CKAppliedComponentAnimationMap &)map
+{
+  [CATransaction begin];
+  CKComponentAnimationID animationID = _nextAnimationID++;
+  [CATransaction setCompletionBlock:^() {
+    auto it = map.find(animationID);
```

Comment:
alas, this won't work because you're capturing a C++ reference in a block. that's akin to capturing a raw pointer; if the component controller deallocates before the block is executed, you'll dereference garbage memory and crash.

I'm a bit flummoxed (as i imagine you are) by how to make this code reusable. Hrm.